pkg/metrics/metrics.go

```diff
@@ -48,6 +44,16 @@ type PrometheusNodeConfig struct {
 	namespace string
 }
 
+// MetricsConfig holds the information required to interact with metrics objects
+type MetricsConfig struct {
```

Comment:
Is there a reason for the `Metrics` prefix? I suggest going with `Config` given the name of the package is `metrics`. We should always ensure there is no stuttering in our APIs. `metrics.MetricsConfig` vs `metrics.Config`.